spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/reactive/LoadBalancerRetryPolicy.java

```diff
@@ -31,28 +32,78 @@
 	 * Return <code>true</code> to retry on the same service instance.
 	 * @param context the context for the retry operation
 	 * @return true to retry on the same service instance
+	 * @deprecated Deprecated in favor of
+	 * {@link #canRetrySameServiceInstance(String, LoadBalancerRetryContext)}
 	 */
+	@Deprecated
 	boolean canRetrySameServiceInstance(LoadBalancerRetryContext context);
 
+	/**
+	 * Return <code>true</code> to retry on the same service instance.
+	 * @param serviceId the serviceId for the retry operation
+	 * @param context the context for the retry operation
+	 * @return true to retry on the same service instance
+	 */
+	default boolean canRetrySameServiceInstance(String serviceId, LoadBalancerRetryContext context) {
```

Comment:
Why couldn't `LoadBalancerRetryContext` have the serviceId so all these methods won't need to change

Reply:
`LoadBalancerRetryContext` is created in `ReactorLoadBalancerClientAutoConfiguration`. At that time we don't have information about service names. So I decided to add new methods.

Reply:
Yes, we don't have it there.